zokrates_cli/src/ops/generate_smtlib2.rs

```diff
@@ -0,0 +1,64 @@
+use crate::constants::{FLATTENED_CODE_DEFAULT_PATH, SMTLIB2_DEFAULT_PATH};
+use clap::{App, Arg, ArgMatches, SubCommand};
+use std::fs::File;
+use std::io::{BufReader, Write};
+use std::path::Path;
+use zokrates_core::ir;
+use zokrates_core::ir::smtlib2::SMTLib2Display;
+use zokrates_core::ir::ProgEnum;
+use zokrates_field::Field;
+
+pub fn subcommand() -> App<'static, 'static> {
+    SubCommand::with_name("generate-smtlib2")
+        .about("Outputs the constraint system in the SMTLib2 format")
+        .arg(
+            Arg::with_name("input")
+                .short("i")
+                .long("input")
+                .help("Path of the binary")
+                .value_name("FILE")
+                .takes_value(true)
+                .required(false)
+                .default_value(FLATTENED_CODE_DEFAULT_PATH),
+        )
+        .arg(
+            Arg::with_name("output")
+                .short("o")
+                .long("output")
+                .help("Path of the output file")
+                .value_name("FILE")
+                .takes_value(true)
+                .required(false)
+                .default_value(SMTLIB2_DEFAULT_PATH),
+        )
+}
+
+pub fn exec(sub_matches: &ArgMatches) -> Result<(), String> {
+    // read compiled program
+    let path = Path::new(sub_matches.value_of("input").unwrap());
+    let file =
+        File::open(&path).map_err(|why| format!("Could not open {}: {}", path.display(), why))?;
+
+    let mut reader = BufReader::new(file);
+
+    match ProgEnum::deserialize(&mut reader)? {
+        ProgEnum::Bn128Program(p) => cli_compute(p, sub_matches),
+        ProgEnum::Bls12_377Program(p) => cli_compute(p, sub_matches),
+        ProgEnum::Bls12_381Program(p) => cli_compute(p, sub_matches),
+        ProgEnum::Bw6_761Program(p) => cli_compute(p, sub_matches),
+    }
+}
+
+fn cli_compute<T: Field>(ir_prog: ir::Prog<T>, sub_matches: &ArgMatches) -> Result<(), String> {
```

Comment:
For consistency with other commands, this should probably be `cli_smtlib2` or something